File: app/javascript/channels/comments_channel.js
import consumer from "./consumer"

consumer.subscriptions.create(
  {
    channel: "CommentsChannel",
    category_name: document.body.dataset.categoryName,
    product_name: document.body.dataset.productName
  },
  {
    received(data) {
      const commentsContainer = document.getElementById("comments")
      if (commentsContainer) {
        commentsContainer.insertAdjacentHTML('beforeend', data.comment)
      }
    }
  }
)

File: app/channels/comments_channel.rb
class CommentsChannel < ApplicationCable::Channel
  def subscribed
     # stream_from "some_channel"
     stream_from "comments_#{params['category_name']}_#{params['product_name']}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
